Python/sysmodule.c

```diff
@@ -2810,7 +2815,8 @@ set_flags_from_config(PyInterpreterState *interp, PyObject *flags)
     SetFlag(preconfig->utf8_mode);
     SetFlag(config->warn_default_encoding);
     SetFlagObj(PyBool_FromLong(config->safe_path));
-    SetFlag(_Py_global_config_int_max_str_digits);
+    SetFlag(config->int_max_str_digits == _PY_LONG_DEFAULT_MAX_STR_DIGITS ?
```

Comment:
This is due to one consequence of this change... Using PyConfig as the canonical store of the actual live value means that it will never contain `-1` at runtime and all API calls to `sys.set_int_max_str_digits()` update the PyConfig structure that is behind sys.flags.  But given we have a tiny amount of logic in here above for the hash_seed setting, I figure this is a fine place for such small logic even if annoying.

We do "lose" information vs what 3.11 and earlier have: we have no indicator as to if a limit was explicitly set or if the default was used as "config" being the original python config at (sub)interpreter start time vs "interpreter state" no longer has a distinction for `int_max_str_digits`.

Is that a problem in practice? I hope not... thoughts?

Reply:
I can't think of a reason why it would be important to differentiate the source of the active limit.